packages/powersync/lib/src/web/sync_worker.dart

```diff
@@ -0,0 +1,258 @@
+/// This file needs to be compiled to JavaScript with the command
+/// dart compile js -O4 packages/powersync/lib/src/web/sync_worker.worker.dart -o assets/db_worker.js
```

Comment:
There is a small typo in this command. With the current file structure and default worker path it should be:
```
dart compile js -O4 packages/powersync/lib/src/web/sync_worker.dart -o assets/powersync_sync_worker.js
```
We currently compile the DB worker [here](https://github.com/powersync-ja/powersync.dart/blob/5dfd36c7dd69f62997504a4b946e301daa24b7a0/scripts/compile_webworker.dart#L10). I think the sync worker should also be compiled there.

We also need to add this file to our Github releases workflow [here](https://github.com/powersync-ja/powersync.dart/blob/5dfd36c7dd69f62997504a4b946e301daa24b7a0/.github/workflows/release.yml#L39).

We have a small CLI tool which devs can use to download a precompiled worker. This sync worker's release file should be added to the [tool](https://github.com/powersync-ja/powersync.dart/blob/5dfd36c7dd69f62997504a4b946e301daa24b7a0/packages/powersync/bin/setup_web.dart#L57).

Reply:
@simolus3 You can leave these changes to me. I'll add the respective changes after this PR is merged. This is mainly to prevent merge conflicts with #194 which refactors some of the powersync files to a new package.

Reply:
Thanks!